### PR TITLE
common: Do not use non-portable constants in mutex_debug

### PR DESCRIPTION
**Pull request by adamemerson**
The mutex initializers in Linux are non-portable, as evidenced by their names
ending with _NP.

Use the POSIX compliant ways to initialize mutices instead.

Signed-off-by: Adam C. Emerson <aemerson@redhat.com>

### CODE REVIEW

File: src/spdk [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 9322c258084c6abdeefe00067f8b310a6e0d9a5a
+Subproject commit c02b179490123a3212b0c0d23b69da13965d1552
```
Comment (cbodley) [inline src/spdk]: unrelated?

Comment (adamemerson) [inline src/spdk]: Well, that's embarrassing.

Fixed.

**Comment by wjwithagen** [comment]
I would assumed that NP would stand for NON-POSIX, but then non-portable would be a nice one as well.
Going to add it in my wip, and see what  FreeBSD thinks of it.

**Comment by wjwithagen** [comment]
LGTM.

FreeBSD compiles, and unittest runs with errors

**Comment by adamemerson** [comment]
What kind of errors? Are you getting them in the mutex_debug unittest?

**Comment by wjwithagen** [comment]
@adamemerson 

LGTM

Sorry Adam.
NO Errors,
